### PR TITLE
Decision strategy: incorporate arrays.

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/theory/arrays/theory_arrays.h [hunk 1]
```diff
@@ -455,6 +453,27 @@ class TheoryArrays : public Theory {
   /** An equality-engine callback for proof reconstruction */
   ArrayProofReconstruction d_proofReconstruction;
 
+  /** The decision strategy for the theory of arrays */
```
Comment (HanielB) [inline src/theory/arrays/theory_arrays.h]: say how it operates

Comment (ajreynol) [reply src/theory/arrays/theory_arrays.h]: Unfortunately I don't know how it does (this is Clark's code).

Comment (barrettcw) [reply src/theory/arrays/theory_arrays.h]: If the "arrays-eager-index" option is enabled, then whenever a read-over-write lemma is generated, a decision request is also generated for the comparison between the indexes that appears in the lemma.

Comment (ajreynol) [reply src/theory/arrays/theory_arrays.h]: I'll add this.

**Pull request merged**